Libs/Professions/professions.lua

```diff
@@ -0,0 +1,137 @@
+            set = function(info, value)
+                Professions:updateOptions(value);
+            end,
+        },
+        selectsubprof = {
+            order = 3,
+            type = "select",
+            name = "Category",
+            values = subProfessionValues;
+            get = function(info)
+                return selectedSubProfession;
+            end,
+            set = function(info, value)
+                Professions:updateSubOptions(value);
+            end,
+        },
+        professors = {
+            order = 4,
+            type = "description",
+            name = function() return Professions:updateProfessors() end;
+        }
+    }
+}
+
+function Professions:updateOptions(value)
+    selectedProfession = value;
+    selectedSubProfession = "";
+    count = #subProfessionValues
+    for i=0, count do subProfessionValues[i]=nil end
+    if value == "Alchemy" then
+        table.insert(subProfessionValues, "Flasks");
+        table.insert(subProfessionValues, "Elixirs");
+        table.insert(subProfessionValues, "Resist Potions");
+        table.insert(subProfessionValues, "Miscellaneous");
+    elseif value == "Blacksmithing" then
+        table.insert(subProfessionValues, "Armor");
+        table.insert(subProfessionValues, "Weapon");
+        table.insert(subProfessionValues, "Miscellaneous");
+    elseif value == "Enchanting" then
+        table.insert(subProfessionValues, "Boots");
+        table.insert(subProfessionValues, "Bracer");
+        table.insert(subProfessionValues, "Chest");
+        table.insert(subProfessionValues, "Cloak");
+        table.insert(subProfessionValues, "Gloves");
+        table.insert(subProfessionValues, "Shield");
+        table.insert(subProfessionValues, "Weapon");
+        table.insert(subProfessionValues, "Oils");
+    elseif value == "Leatherworking" then
+        table.insert(subProfessionValues, "Armor");
+        table.insert(subProfessionValues, "Cloak");
+    elseif value == "Tailoring" then
+        table.insert(subProfessionValues, "Bags");
+        table.insert(subProfessionValues, "Armor");
+        table.insert(subProfessionValues, "Cloak");
+    end
+end
+
+function Professions:updateSubOptions(value)
+    selectedSubProfession = value;
+end
+
+function Professions:updateProfessors()
+    local professors = "";
+    if selectedSubProfession ~= "" then
+        local args = { 
+            string.lower(subProfessionValues[selectedSubProfession]),
+            "all"
+        };
+        professors = Professions:getProfessors(args);
+    end
+    return professors;
+end
+
+function Professions:parseInput(args)
+    local slot = _G[selectedProfession]:parseSlot(table.remove(args, 1));
+    local caty = _G[selectedProfession]:parseCaty(table.remove(args, 1));
+    return { slot, caty };
+end
+
+function Professions:getProfessors(args)
+    local input = self:parseInput(args);
+    local slot = table.remove(input, 1);
+    local caty = table.remove(input, 1);
```

Comment:
I am trying to understand these 3 lines.     `args` object is somthing like this right
```
{
        distilledwisdom = {'Shamattack'},
        supremepower = {'shamattack'},
        titans = {},
        "all"
}
```

if we want to get any element from there we can use the `select` function instead of removing and altering the original object
```
element = select(<position index>, <table>)
first_element = select(1, args)
```
but yeah I'm still confused

Reply:
args would be something like { 'flask', 'all' } or { 'flask', 'wisdom' }
The purpose of this is to filter input. Select might work better here rather than table.remove. Either way it doesn't matter too much since remove is just popping off the top which is a fast operation as well. The table is modified locally but not reused so a pop or select is fine.

Reply:
Maybe I can be clearer since reading what I wrote back to myself is confusing hahaha.

Lets say you wanted to find all Flask recipes and who makes them. The code path would generate args like this:
```
args = {
    'flask',
    'all'
};
```
The reason why this is a pairing of values is to filter it down further if needed, such as:
```
args = {
    'flask',
    'distilledwisdom'
};
```
Hope that makes a little more sense. I can add some comments in the code here so we don't get confused in the future

Reply:
Looking into this more I'm not sure WoW Lua `select(index, ...)` will work. Here is a debug dump of what returns when I do a select versus a table.remove:
![Screenshot_2020-11-28_13-01-35](https://user-images.githubusercontent.com/16396585/100522824-c58ba800-3179-11eb-9467-ffa8bce15960.png)
Since select returns an entire list that isn't quite what we want here. Now it could be simplified with a basic indexing:
```
 90     local slot = input[1];
 91     local caty = input[2];
```
Not sure why i didn't do this before haha